### DISCUSSION TITLE
Urdu Translation

### DISCUSSION THREAD

**Discussion by AhsanSheraz** [fastapi/fastapi#9973] [Translations]
Welcome to the issue that coordinates the Urdu translation effort.

Purpose
Avoiding several people working on the same document at the same time.
The first person who declares that he/she is working on a translation gets the responsibility to carry it out. If a PR seems to be stalled, we can discuss a transfer of responsibility here.

Enforcing best practices
Best practices are listed later in this description. You can propose your practice at any time, ideally with a supporting source and an example.

Defining and sharing best practices will help to avoid common mistakes and will allow faster and easier reviews.

Help and build the community
Do not hesitate to ask any questions regarding the Urdu translation effort here. The stronger the community, the more effective we will be and the more we will enjoy.

Provide an Urdu translation for this awesome library (last but not least)
If you are here, you probably like FastAPI, and maybe you even speak Urdu. Giving more people the opportunity to get started using the documentation in its native language will encourage adoption. In that spirit, let's contribute to the magic of open source in this way.

How to contribute
Review
Keep in mind that the easiest way to participate is to review the PRs. We need to avoid accumulating PRs waiting for review.

Translate
If you are not familiar with contributing to open-source projects have a look at https://github.com/firstcontributions/first-contributions.

In any case, take a look at the documentation section related to the [contribution](https://fastapi.tiangolo.com/contributing/#development-contributing) and more precisely the part about the [documentation](https://fastapi.tiangolo.com/contributing/#docs).

Once you are decided to translate a document, make yourself known here by leaving a message here (eg. https://github.com/tiangolo/fastapi/discussions/9973).

Organize
If you wish, your energy is welcome to help with the organization. Bringing together motivated people and helping them get the job done is essential. Moreover, we can surely learn a lot from the translation work of other languages that are much more advanced and we can have a significant impact if we put good processes in place that can help the whole community

Links for Urdu translation:
https://translatiz.com/translation/english-to-urdu
https://lingvanex.com/translation/english-to-urdu

Also, if the English document got updated, we just have to update the first commit and the conflicts will reveal an updated part of the document 🪄

**Comment by AhsanSheraz** [comment]
I am currently working on benchmarks page translation

**Reply by AhsanSheraz** [reply]
Here is the [MR](https://github.com/tiangolo/fastapi/pull/9974) related to benchmark translation

**Reply by AhsanSheraz** [reply]
URL to review this translation:
[Orignal Page](https://64c8bb315b09554ce9eef008--fastapi.netlify.app/benchmarks/)
[Translated Page](https://64c8bb315b09554ce9eef008--fastapi.netlify.app/ur/benchmarks/)

**Reply by AhsanSheraz** [reply]
MR is merged

**Comment by AhsanSheraz** [comment]
@tiangolo can you please add labels to this conversation as well...

**Reply by tiangolo** [reply]
Yep, done! 🚀

**Comment by tiangolo** [comment]
~There's a new translation PR to be reviewed: #9974 by @AhsanSheraz~ Good job! This is done. 🍰☕

**Reply by talhaumer** [reply]
Approved

**Comment by AhsanSheraz** [comment]
Added translation for external-links page and fix the heading location benchmarks page for urdu translation. Here is the [MR](https://github.com/tiangolo/fastapi/pull/10046)

**Reply by AhsanSheraz** [reply]
@talhaumer please review this MR. here is the [translated page](https://c1c699fb.fastapitiangolo.pages.dev/ur/external-links/) and [original page](https://c1c699fb.fastapitiangolo.pages.dev/external-links/)

**Reply by AhsanSheraz** [reply]
@tiangolo fixed the issues, you can take a look. Everything is ok.

**Comment by tiangolo** [comment]
~There's a new translation PR to be reviewed: #10046 by @AhsanSheraz~ Good job! This is done. 🍰☕

**Comment by MuhammadAbbas010** [comment]
@AhsanSheraz @talhaumer seeing that the old Urdu documentation has been removed for some time now, I would like to take this work forward. I've generated the LLM-prompt.md as well as a test-LLM.md file. If you could, please provide some feedback on it before I start translating the entire docs? 

[FastAPI Fork](https://github.com/MuhammadAbbas010/fastapi/tree/master/docs/ur)

tq

**Reply by YuriiMotov** [reply]
Please see: https://fastapi.tiangolo.com/translations/#request-a-new-language

We no longer accept translation PRs.
If we decide to support this language, we will create the prompt together and then we will generate translations automatically and ask native speakers to review them

**Reply by MuhammadAbbas010** [reply]
Hi Yurii thanks for the info. Your message makes it seem like the fastapi team decides when a language reaches enough popularity to be translated, but the docs say that if I gather atleast 3 people the process can begin, may I know which one is it? 

If it ever does happen, I will gladly review the translations :)

**Reply by YuriiMotov** [reply]
Hi!

The process is:
* You need to find 2 more native speakers who are ready to help with reviewing the translations
* Then Sebastian will consider adding this language
* If Sebastian approves adding this language, I will add it and organize the translation process
